sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -149,6 +150,15 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
             Duration retryAfter = ImplUtils.getRetryAfterFromHeaders(response.getHeaders(), OffsetDateTime::now);
             return PollingUtils.convertResponse(response.getValue(), serializer, pollResponseType)
                 .map(value -> new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, value, retryAfter))
+                .onErrorResume(error ->
+                    error instanceof UncheckedIOException,
+                    error -> {
```

Comment:
Each service operation either uses the old polling pattern or the new pattern. Is this something we can identify when creating the polling strategy? Ignoring errors to distinguish old and new LRO patterns may lead to suppressing other issues.

Reply:
It is pretty hard at runtime.

The response of PUT in old is the poll result:
https://github.com/Azure/azure-sdk-for-java/blob/69cf792e349b30e84fb1a65a56b6f673f625eb05/sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java#L205

The response of PUT in new is the Resource:
https://github.com/Azure/azure-sdk-for-java/blob/69cf792e349b30e84fb1a65a56b6f673f625eb05/sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java#L252

We likely not able to distinguish it only from runtime behavior.

It is possible to deduce from TypeSpec in codegen time, and use e.g. `OperationResourceNewPollingStrategy` when it is "Azure.Core.LongRunningResourceCreateOrReplace".
However it won't solve for DPG (if dev later add the model to LRO).